RecoHI/HiEvtPlaneAlgos/interface/HiEvtPlaneFlatten.h

```diff
@@ -37,108 +44,314 @@ class HiEvtPlaneFlatten {
     }
-    for(int i = 0; i<hbins; i++) {
-      flatX[i]=0;
-      flatY[i]=0;
-      flatCnt[i]=0;
-    } 
+    for(int i = 0; i<hbins_; i++) {
+      flatX_[i]=0;
+      flatY_[i]=0;
+      flatXDB_[i]=0;
+      flatYDB_[i]=0;
+      flatCnt_[i]=0;
+    }
+    for(int i = 0; i<obins_; i++) {
+      xoff_[i]=0;
+      yoff_[i]=0;
+      xoffDB_[i]=0;
+      yoffDB_[i]=0;
+      xyoffcnt_[i]=0;
+      xyoffmult_[i]=0;
+      pt_[i]=0;
+      pt2_[i]=0;
+      ptDB_[i]=0;
+      pt2DB_[i]=0;
+      ptcnt_[i]=0;
+    }
+  }
+
+  int getCutIndx(int centbin, double vtx, int iord) const
+  {
+    int cut;
+    if(centbin < 0 ) return -1;
+    int ibin = centbin;
+    int ivtx = (vtx-minvtx_)/delvtx_;
+    if(vtx < minvtx_ || ivtx >= nvtxbins_) return -1;
+    cut = hOrder_*nvtxbins_*ibin + hOrder_*ivtx + iord;
+    if(cut<0 || cut>=hbins_) return -1;
+    return cut;
   }
 
-  int GetCutIndx(int centbin, double vtx, int iord)
+  int getOffsetIndx(int centbin, double vtx) const
   {
     int cut;
-    int icent = centbin/centbinComp;
-    if(icent < 0 || icent > hcentbins) return -1;
-    int ivtx = (vtx-minvtx)/delvtx;
-    if(ivtx < 0 || ivtx > nvtxbins) return -1;
-    cut = hOrder*nvtxbins*icent + hOrder*ivtx + iord;
-    if(cut<0 || cut>hbins) return -1;
+    if(centbin < 0 ) return -1;
+    //int ietbin = hfetbins*log10( 9.*(et/scale)+1.);
+    //if(ietbin>hfetbins) ietbin=hfetbins-1;
+    int ibin = centbin;
+    int ivtx = (vtx-minvtx_)/delvtx_;
+    if(ivtx < 0 || ivtx > nvtxbins_) return -1;
+    cut = nvtxbins_*ibin + ivtx ;
+    if(cut<0 || cut>hbins_) return -1;
     return cut;
   }
-  
-  void Fill(double psi, double vtx, int centbin)
+
+  void fill(double psi, double vtx, int centbin)
   {
     if(fabs(psi)>4 ) return;
-    for(int k = 0; k<hOrder; k++) {
-      double fsin = sin(vorder*(k+1)*psi);
-      double fcos = cos(vorder*(k+1)*psi);
-      int indx = GetCutIndx(centbin,vtx,k);
+    for(int k = 0; k<hOrder_; k++) {
+      double fsin = sin(vorder_*(k+1)*psi);
+      double fcos = cos(vorder_*(k+1)*psi);
+      int indx = getCutIndx(centbin,vtx,k);
       if(indx>=0) {
-	flatX[indx]+=fcos;
-	flatY[indx]+=fsin;
-	++flatCnt[indx];
+	flatX_[indx]+=fcos;
+	flatY_[indx]+=fsin;
+	++flatCnt_[indx];
       }
     }
   }
+  void fillOffset(double s, double c, uint m, double vtx, int centbin)
+  {
+    int indx = getOffsetIndx(centbin,vtx);
+    if(indx>=0) {
+      xoff_[indx]+=c;
+      yoff_[indx]+=s;
+      xyoffmult_[indx]+=m;
+      ++xyoffcnt_[indx];
+    }
+  }
+  void fillPt(double ptval, double vtx, int centbin)
+  {
+    int indx = getOffsetIndx(centbin,vtx);
+    if(indx>=0) {
+      pt_[indx]+=ptval;
+      pt2_[indx]+=ptval*ptval;
+      ++ptcnt_[indx];
+    }
+  }
+
+  void setCaloCentRefBins(const int caloCentRefMinBin, const int caloCentRefMaxBin) {
+    caloCentRefMinBin_ = caloCentRefMinBin;
+    caloCentRefMaxBin_ = caloCentRefMaxBin;
+    caloCentRefVal_ = 1.;
+  }
+
+  double EtScale(double vtx, int centbin) {
```

Comment:
(method names should start with a lower case letter, hence, "EtScale" -> "etScale"